patchwork-capabilities/src/main/java/com/patchworkmc/impl/capability/CapabilityProviderInterface.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.impl.capability;
+
+import javax.annotation.Nonnull;
+
+import net.minecraftforge.common.capabilities.CapabilityProvider;
+
+/**
+ * A holder for {@link CapabilityProvider}
+ */
+public interface CapabilityProviderInterface {
+
+	@Nonnull
+	CapabilityProvider getCapabilityProvider$impl();
```

Comment:
I don't think you can have a raw $ in a method declaration...

Reply:
You can, and it's completely valid.

Reply:
Why is there an impl method in an interface?

Reply:
The interface was originally meant to be a proxy to all public methods in CapabilityProvider but I decided to create a proxy class instead. It still shouldn’t be used. I’ll rename the method.